x/gamm/pool-models/concentrated-liquidity/pool.go

```diff
@@ -10,6 +12,67 @@ var (
 	_ types.PoolI = &Pool{}
 )
 
+func NewConcentratedLiquidityPool(poolId uint64) (Pool, error) {
+	poolAddr := types.NewPoolAddress(poolId)
+
+	// pool thats created up to ensuring the assets and params are valid.
+	// We assume that FuturePoolGovernor is valid.
+	pool := &Pool{
+		Address: poolAddr.String(),
+		Id:      poolId,
+	}
+
+	return *pool, nil
+}
+
+// liquidity0 takes an amount of asset0 in the pool as well as the sqrtpCur and the nextPrice
+// pa is the smaller of sqrtpCur and the nextPrice
+// pb is the larger of sqrtpCur and the nextPrice
+func liquidity0(amount int64, pa, pb sdk.Dec) sdk.Dec {
+	if pa.GT(pb) {
+		pa, pb = pb, pa
+	}
+	product := pa.Mul(pb).Quo(sdk.NewDec(10).Power(6))
```

Comment:
let's figure out powers

Reply:
This is to be done in future reviews to avoid this PR being sole blocker for future works